### ISSUE TITLE
local-gadget trace tcp ignores SIGINT if started before container on WSL2

### ISSUE THREAD

**Issue by wedaly** [kinvolk/inspektor-gadget#958]
## Description

On Ubuntu 22.04 WSL2, running `local-gadget trace tcp` with no other containers running I noticed that:
1. CTRL-C does not cause local-gadget to terminate
2. Docker containers started _after_ `local-gadget` do not appear in the TCP trace output.

## Impact

Have to `kill -9` to exit local-gadget. 

## Environment and steps to reproduce

1. **Set-up**:
 * Running WSL2 Ubuntu 22.04 on Windows 11. `uname -r` shows "5.10.102.1-microsoft-standard-WSL2".
 * I installed docker community edition (NOT Docker Desktop). This is the output of `docker version`:
 ```
 docker version
Client: Docker Engine - Community
 Version:           20.10.18
 API version:       1.41
 Go version:        go1.18.6
 Git commit:        b40c2f6
 Built:             Thu Sep  8 23:11:43 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.18
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.6
  Git commit:       e42327a
  Built:            Thu Sep  8 23:09:30 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
  ```
* Built local-gadget from source on commit 4685156ce3ab397d31191d3a28dbaba8cae4a6fc using `make local-gadget`

3. **Task**: trying to run `local-gadget trace tcp` on a busybox container making an HTTP request to example.com
4. **Action(s)**:
  a. Start the tracer: `sudo ./local-gadget trace tcp`
  b. Start the docker container: `docker run -it --rm --name test-container busybox /bin/sh`
  c. Make the HTTP request in the busybox container: `wget http://example.com`
  d. Attempt to exit local-gadget with CTRL-C.
5. **Error**:
  a. local-gadget does not terminate.
  b. No TCP traces appear in local-gadget output
  
<img width="939" alt="image" src="https://user-images.githubusercontent.com/2948394/191129155-09f974b8-5aa0-4f7f-966b-61326bd9bc80.png">

Note that if the busybox container starts BEFORE local-gadget, then I see the TCP trace show up and I'm able to exit local-gadget using CTRL-C.

Also, if I exit the busybox container, then local-gadget also exits.

## Expected behavior

1. TCP trace appears in local-gadget output.
2. CTRL-C exits the local-gadget process.

## Additional information

* I was unable to reproduce this on Fedora 36 Workstation.
* Running local-gadget with the `-v` flag, I see:
```
sudo ./local-gadget trace tcp -v
WARN[0000] Runtime enricher (containerd): couldn't get current containers
WARN[0000] Runtime enricher (cri-o): couldn't get current containers
DEBU[0000] Checking "/usr/bin/runc": done
DEBU[0000] Checking "/usr/sbin/runc": fanotify: mark error, no such file or directory
DEBU[0000] Checking "/usr/local/sbin/runc": fanotify: mark error, no such file or directory
DEBU[0000] Checking "/run/torcx/unpack/docker/bin/runc": fanotify: mark error, no such file or directory
DEBU[0000] Checking "/host/usr/bin/runc": fanotify: mark error, no such file or directory
DEBU[0000] Checking "/host/usr/sbin/runc": fanotify: mark error, no such file or directory
DEBU[0000] Checking "/host/usr/local/sbin/runc": fanotify: mark error, no such file or directory
DEBU[0000] Checking "/host/run/torcx/unpack/docker/bin/runc": fanotify: mark error, no such file or directory
```

* Built with debug symbols and used `dlv` to attach to the local-gadget process while it was hung:
<img width="1112" alt="image" src="https://user-images.githubusercontent.com/2948394/191133197-5f5bb130-7872-4717-805f-9d78284ca7ec.png">

* When I inspect goroutine 1 (blocking on `sync.runtime_Semacquire`) and navigate up frames until I reach inspektor gadget code, I see this call stack:
```
Frame 5: /go/src/github.com/kinvolk/inspektor-gadget/pkg/runcfanotify/runcfanotify.go:579
Frame 6: /go/src/github.com/kinvolk/inspektor-gadget/pkg/container-collection/options.go:583
Frame 7: /go/src/github.com/kinvolk/inspektor-gadget/pkg/container-collection/container-collection.go:343
Frame 8: /go/src/github.com/kinvolk/inspektor-gadget/pkg/local-gadget-manager/local-gadget-manager.go:368
Frame 9: /go/src/github.com/kinvolk/inspektor-gadget/cmd/local-gadget/trace/trace.go:50
```
* Inspecting goroutine 15 (`syscall.Syscall`), I see this call stack:
```
Frame 11: /go/pkg/mod/github.com/s3rj1k/go-fanotify/fanotify@v0.0.0-20210917134616-9c00a300bb7a/fanotify.go:194
Frame 12: /go/src/github.com/kinvolk/inspektor-gadget/pkg/runcfanotify/runcfanotify.go:339
Frame 13: /go/src/github.com/kinvolk/inspektor-gadget/pkg/runcfanotify/runcfanotify.go:465
```

**Comment by alban** [comment]
I am unfamiliar with WSL2 and Docker on Windows. Are local-gadget and the docker server running on the same kernel? Since local-gadget uses fanotify to detect when runc is executed, they would need to run on the same kernel (and not merely share the same filesystem).

**Comment by wedaly** [comment]
Tracked it down with a lot of `log.Infof` statements. This was caused by the symlink from `/var/run` to /run` on WSL2 Ubuntu 22.04. Proposed fix in #963
